# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/firefox/hotfix-arm64-no-sve.patch
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

 0:38.76 In file included from /srv/t2/src.firefox.arm64.240520.123212.30252/firefox-126.0/media/libvpx/libvpx/vp9/encoder/arm/neon/vp9_error_sve.c:17:
 0:38.76 /srv/t2/src.firefox.arm64.240520.123212.30252/firefox-126.0/media/libvpx/libvpx/vpx_dsp/arm/vpx_neon_sve_bridge.h:16:10: fatal error: arm_neon_sve_bridge.h: No such file or directory
 0:38.76    16 | #include <arm_neon_sve_bridge.h>
 0:38.76       |          ^~~~~~~~~~~~~~~~~~~~~~~

--- firefox-126.0/media/libvpx/moz.build.vanilla	2024-05-20 12:42:59.564961088 +0200
+++ firefox-126.0/media/libvpx/moz.build	2024-05-20 12:52:49.667985633 +0200
@@ -81,7 +81,10 @@
     SOURCES += [ '/media/libvpx/config/win/aarch64/vpx_config.c' ]
 elif CONFIG['TARGET_CPU'] == 'aarch64':
     EXPORTS.vpx += files['ARM64_EXPORTS']
-    SOURCES += files['ARM64_SOURCES']
+    # We don't have gcc-14 w/ sve bridge yet
+    SOURCES += [
+            f for f in files['ARM64_SOURCES'] if not f.endswith('_sve.c')
+        ]
     ASFLAGS += [ '-I%s/media/libvpx/config/linux/arm64/' % TOPSRCDIR ]
     LOCAL_INCLUDES += [ '/media/libvpx/config/linux/arm64/' ]
     SOURCES += [ '/media/libvpx/config/linux/arm64/vpx_config.c' ]
--- firefox-126.0/media/libvpx/config/linux/arm64/vpx_config.asm.vanilla	2024-05-20 13:28:07.489268565 +0200
+++ firefox-126.0/media/libvpx/config/linux/arm64/vpx_config.asm	2024-05-20 13:28:15.817269230 +0200
@@ -12,8 +12,8 @@
 .equ HAVE_NEON ,  1
 .equ HAVE_NEON_DOTPROD ,  1
 .equ HAVE_NEON_I8MM ,  1
-.equ HAVE_SVE ,  1
-.equ HAVE_SVE2 ,  1
+.equ HAVE_SVE ,  0
+.equ HAVE_SVE2 ,  0
 .equ HAVE_MIPS32 ,  0
 .equ HAVE_DSPR2 ,  0
 .equ HAVE_MSA ,  0
--- firefox-126.0/media/libvpx/config/linux/arm64/vpx_config.h.vanilla	2024-05-20 13:29:09.291273503 +0200
+++ firefox-126.0/media/libvpx/config/linux/arm64/vpx_config.h	2024-05-20 13:29:14.969273956 +0200
@@ -21,8 +21,8 @@
 #define HAVE_NEON 1
 #define HAVE_NEON_DOTPROD 1
 #define HAVE_NEON_I8MM 1
-#define HAVE_SVE 1
-#define HAVE_SVE2 1
+#define HAVE_SVE 0
+#define HAVE_SVE2 0
 #define HAVE_MIPS32 0
 #define HAVE_DSPR2 0
 #define HAVE_MSA 0
--- firefox-126.0/media/libvpx/config/linux/arm64/vp9_rtcd.h.vanilla	2024-05-20 13:29:23.514274639 +0200
+++ firefox-126.0/media/libvpx/config/linux/arm64/vp9_rtcd.h	2024-05-20 13:30:05.799278017 +0200
@@ -99,9 +99,9 @@
     (void)flags;
 
     vp9_block_error = vp9_block_error_neon;
-    if (flags & HAS_SVE) vp9_block_error = vp9_block_error_sve;
+    //if (flags & HAS_SVE) vp9_block_error = vp9_block_error_sve;
     vp9_block_error_fp = vp9_block_error_fp_neon;
-    if (flags & HAS_SVE) vp9_block_error_fp = vp9_block_error_fp_sve;
+    //if (flags & HAS_SVE) vp9_block_error_fp = vp9_block_error_fp_sve;
 }
 #endif
 
--- firefox-126.0/media/libvpx/config/linux/arm64/vpx_dsp_rtcd.h.vanilla	2024-05-20 15:12:58.414011538 +0200
+++ firefox-126.0/media/libvpx/config/linux/arm64/vpx_dsp_rtcd.h	2024-05-20 15:13:28.707013958 +0200
@@ -1150,7 +1150,7 @@
     vpx_sse = vpx_sse_neon;
     if (flags & HAS_NEON_DOTPROD) vpx_sse = vpx_sse_neon_dotprod;
     vpx_sum_squares_2d_i16 = vpx_sum_squares_2d_i16_neon;
-    if (flags & HAS_SVE) vpx_sum_squares_2d_i16 = vpx_sum_squares_2d_i16_sve;
+    //if (flags & HAS_SVE) vpx_sum_squares_2d_i16 = vpx_sum_squares_2d_i16_sve;
     vpx_variance16x16 = vpx_variance16x16_neon;
     if (flags & HAS_NEON_DOTPROD) vpx_variance16x16 = vpx_variance16x16_neon_dotprod;
     vpx_variance16x32 = vpx_variance16x32_neon;
